docs/adrs/0005-allow-sub-templates.md

```diff
@@ -0,0 +1,86 @@
+# 1. Allow sub-templates
+
+Date: 2022/09/27
+
+## Status
+
+Proposed.
+
+## Context
+
+As components become larger (for example, because you are implementing a whole page), it becomes
+useful to be able to extract sections of the view to a different file. ActionView has
+partials, and ViewComponent lacks a similar mechanism.
+
+ActionView partials have the problem that their interface isn't introspectable. Data
+may be passed into the partial via ivars or locals, and it is impossible to know
+which without actually opening up the file. Additionally, partials are globally
+invocable, thus making it difficult to detect if a given partial is in use or not,
+and who are its users.
```

Comment:
```suggestion
The interface of ActionView partials cannot be introspected. Because data may be passed into the partial via ivars or locals it is impossible to know which without reading the file. Partials are also globally accessible, making it difficult to determine if a given partial is in use or not.
```